### PR TITLE
reloadOnChange cleanup

### PR DESCRIPTION
**Pull request by HaoK**
cc @divega @rustd @phenning

### CODE REVIEW

**Comment by rustd** [comment]
:shipit: once we have a co-herence signed build.

File: src/BaseTemplates/StarterWeb/Startup.cs [hunk 1]
```diff
@@ -19,10 +19,14 @@ public Startup(IHostingEnvironment env, IApplicationEnvironment appEnv)
             // Set up configuration sources.
             var builder = new ConfigurationBuilder()
                 .SetBasePath(appEnv.ApplicationBasePath)
-                .AddJsonFile("appsettings.json")
+                .AddJsonFile(source =>
+                {
+                    source.Path = "appsettings.json";
+                    source.ReloadOnChange = true;
```
Comment (divega) [inline src/BaseTemplates/StarterWeb/Startup.cs:25]: Didn't we create a simple overload for this? Does the simplest/most common usage in our default template require a lambda for this provider while it doesn't for the others? :disappointed:

Comment (rustd) [inline src/BaseTemplates/StarterWeb/Startup.cs:25]: AddJsonFile("appsettings.json", reloadonChange=true) would be better here. no?

**Comment by rustd** [comment]
@HaoK @divega @shirhatti pinging on this. Are we making changes to config APIs to improve this for RC2?

File: src/BaseTemplates/StarterWeb/Startup.cs (hunk 1 above)
Comment (divega) [inline src/BaseTemplates/StarterWeb/Startup.cs:25]: I now remember there was a big discussion around this and ended up keeping the existing overloads but adding the lambda syntax to support the new parameters (including reload-on-change). Given that I believe we shouldn't try to rush any more changes in this area and that this is the best we can do for RC2. But we should take another look early in the next milestone and if possible cleanup and rationalize the overloads to have something that feels more cohesive. I have filed https://github.com/aspnet/Configuration/issues/420.

**Comment by divega** [comment]
@rustd this is ready as far as I am concerned. Should @HaoK push it or should we wait for someone working on templates to do it?

**Comment by rustd** [comment]
@phenning can you please merge this once there is a successful coherence signed builds with these changes.

**Pull request merged**